### ISSUE TITLE
[MLv2] Infinite recursion when building a custom column with joined card columns

### ISSUE THREAD

**Issue by kulyk** [metabase/metabase#33086]
Using joined card columns in a custom column can throw an infinite recursion error:

```
Error calculating metadata for :/: Error calculating metadata for :/: Maximum call stack size exceeded
```

### To Reproduce

1. Checkout the [joins FE branch](https://github.com/metabase/metabase/pull/32912)
2. Run the [E2E test](https://github.com/metabase/metabase/blob/9fef23dc6be04a0b83cd11f9090ee0c71d51b343/e2e/test/scenarios/joins/joins.cy.spec.js#L151)

**Comment by camsaul** [comment]
Duplicate of #32063; this will be fixed by #33016

**Comment by camsaul** [comment]
Duplicate of #32063 

This will be fixed by #33016

**Comment by gfviegas** [comment]
Adding a comment instead of a +1 as this issue is closed and the associated PR merged. 

I'm still experiencing this issue in latest version (0.47.0, free, self-hosted). It happens in more than one question, and the result of the question itself renders fine, but when opening the notebook I see the "something went wrong" page.

When peeking through the console log it shows the referenced error in this issue _(Error calculating metadata for .... : Maximum call stack size exceeded)_

![image](https://github.com/metabase/metabase/assets/6178801/8189280f-4b90-4479-9a4c-83b6269dd844)


Debug info:
```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20+8",
    "os.name": "Linux",
    "os.version": "5.15.0-1030-aws",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "bigquery-cloud-sdk",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.10"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.4"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-08-16",
      "tag": "v0.47.0",
      "branch": "release-x.47.x",
      "hash": "682ec2d"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```
